packages/common/src/store/pages/chat/middleware.ts

```diff
@@ -0,0 +1,73 @@
+import { ChatEvents, type sdk } from '@audius/sdk'
+import { Middleware } from 'redux'
+
+import { actions as chatActions } from './slice'
+
+const {
+  connect,
+  disconnect,
+  addMessage,
+  incrementUnreadCount,
+  setMessageReactionSucceeded
+} = chatActions
+
+export const chatMiddleware =
+  (audiusSdk: () => Promise<ReturnType<typeof sdk>>): Middleware =>
+  (store) => {
+    let messageListener: ChatEvents['message'] | null = null
+    let reactionListener: ChatEvents['reaction'] | null = null
+    let openListener: ChatEvents['open'] | null = null
+    let closeListener: ChatEvents['close'] | null = null
+
+    return (next) => (action) => {
+      if (connect.match(action)) {
```

Comment:
Wonder if we should track some global connection state? i.e. if we hit two connect actions we're going to register double listeners, seems like a bug waiting to happen...

Reply:
Yeah good idea. I originally had something for this back when the chatsApi returned the websocket but forgot to add something back to track